.github/workflows/main_ci.yml

```diff
@@ -0,0 +1,19 @@
+---
+name: Main CI
+on:
+  pull_request:
+    branches: [ main ]
+  merge_group:
+    branches: [ main ]
```

Comment:
Do we want to include the `stable/*` triggers in here as well, so we don't have to remember to update things after new minor releases?